traits/tests/test_has_traits.py

```diff
@@ -604,6 +604,68 @@ class A(HasTraits):
             {"foo": A.class_traits()["foo"]},
         )
 
+    def test_decorated_changed_method(self):
+        # xref: enthought/traits#527
+        # Traits should ignore the _changed magic naming.
+
+        events = []
+
+        class A(HasTraits):
+            foo = Int()
+
+            @on_trait_change("foo")
+            def _foo_changed(self, obj, name, old, new):
+                events.append((obj, name, old, new))
+
+        a = A()
+        a.foo = 23
+        self.assertEqual(
+            events,
+            [(a, "foo", 0, 23)],
+        )
+
+    def test_observed_changed_method(self):
+        events = []
+
+        class A(HasTraits):
+            foo = Int()
+
+            @observe("foo")
+            def _foo_changed(self, event):
+                events.append(event)
+
+        a = A()
+        a.foo = 23
+        self.assertEqual(len(events), 1)
+        event = events[0]
+        self.assertEqual(event.object, a)
+        self.assertEqual(event.name, "foo")
+        self.assertEqual(event.old, 0)
+        self.assertEqual(event.new, 23)
+
+    def test_decorated_changed_method_subclass(self):
+        # xref: enthought/traits#527
+        # Traits should ignore the _changed magic naming.
+
+        events = []
+
+        class A(HasTraits):
+            foo = Int()
+
+            @on_trait_change("foo")
+            def _foo_changed(self, obj, name, old, new):
+                events.append((obj, name, old, new))
+
+        class B(A):
+            pass
```

Comment:
Not sure if it bears mentioning, but #527 was showing an exception at class creation time for this particular case.

Reply:
Yes, for regression test purposes I guess I could have stopped the test at the point of the subclass creation. But it felt natural here to extend the test to make a basic check on the subsequent behaviour.

Reply:
Yes, I was mostly wondering if a comment was needed to show the motivation for future readers.

Reply:
Ah, sorry; I misunderstood your comment. We've got the "xref: enthought/traits#527" comment for this test; I think that's probably good enough.